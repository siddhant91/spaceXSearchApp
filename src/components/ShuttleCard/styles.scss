@import '../../styles/main.scss';
.spacex-shuttle-card {
	background-color: $spacex-color-white;
	min-height: 500px;
	height: fit-content;
	@include padding($spacex-spacing-3, $spacex-spacing-3, $spacex-spacing-3, $spacex-spacing-3);
	width: 100%;
	@include margin($spacex-spacing-2);
	@media only screen and (min-width: $spacex-breakpoint-tablet) {
		@include margin(0, $spacex-spacing-2, $spacex-spacing-2, 0);
		width: 48%;
	}
	@media only screen and (min-width: $spacex-breakpoint-desktop) {
		@include margin(0, $spacex-spacing-2, $spacex-spacing-2, 0);
		width: 24%;
		&:nth-child(4n) {
			margin-right: 0;
		}
	}
	border-radius: $spacex-spacing-1;
	&__image {
		background-color: $spacex-background-grey;
		&--img {
			height: 40%;
			width: 80%;
		}
	}
	&__name {
		color: $spacex-color-primary-green;
		@include margin($spacex-spacing-3, 0, $spacex-spacing-3, 0);
	}
	&__details {
		@include margin($spacex-spacing-2, 0, $spacex-spacing-3, 0);
	}
}
