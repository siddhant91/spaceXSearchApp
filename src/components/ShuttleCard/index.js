import { arrayOf, shape, string } from 'prop-types';

// Styles
import './styles.scss';

function ShuttleCard({ shuttleData }) {
	const {
		missionIds,
		missionName,
		launchLanding,
		launchSuccess,
		launchYear,
		missionPatchImage,
	} = shuttleData;
	const renderMissionIds = () => {
		return (
			<div className="spacex-shuttle-card__mission-id">
				<strong>Mission Ids:</strong>
				<ul>{missionIds && missionIds.length > 0 && missionIds.map((id) => <li>{id}</li>)}</ul>
			</div>
		);
	};

	const getOtherDetails = ({ label, value }) => {
		return (
			<div className="spacex-shuttle-card__details">
				<strong>{label}</strong>
				<span>{value}</span>
			</div>
		);
	};
	return (
		<div className="spacex-shuttle-card" key={`${missionName}-${launchYear}`}>
			<div className="spacex-shuttle-card__image d-flex justify-content-center">
				<img src={missionPatchImage} alt={missionName} />
			</div>
			<h3 className="spacex-shuttle-card__name">{missionName}</h3>
			{renderMissionIds()}
			{getOtherDetails({ label: 'Launch Year:', value: launchYear })}
			{getOtherDetails({ label: 'Successful Launch:', value: launchSuccess })}
			{getOtherDetails({ label: 'Successful Landing:', value: launchLanding })}
		</div>
	);
}

ShuttleCard.propTypes = {
	shuttleData: shape({
		missionIds: arrayOf(string),
		missionName: string,
		launch_landing: string,
		launch_success: string,
		launchYear: string,
		missionPatchImage: string,
	}).isRequired,
};

ShuttleCard.defaultProps = {};

export default ShuttleCard;
