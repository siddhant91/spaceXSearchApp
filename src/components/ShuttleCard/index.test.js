import { render, cleanup, screen } from '@testing-library/react';

import ShuttleCard from './index';

afterEach(cleanup);

const mockData = {
	missionIds: ['1234'],
	missionName: 'FalconSat',
	launchLanding: 'false',
	launchSuccess: 'false',
	launchYear: '2006',
	missionPatchImage: 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
};
const renderComponent = () => {
	return <ShuttleCard shuttleData={mockData} />;
};

it('Matches snapshot', () => {
	const { asFragment } = render(renderComponent());
	expect(asFragment()).toMatchSnapshot();
});

it('renders ShuttleCards list', async () => {
	render(renderComponent());
	const shuttleCards = await screen.findAllByTestId('shuttle-card');
	expect(shuttleCards).toHaveLength(1);
});
