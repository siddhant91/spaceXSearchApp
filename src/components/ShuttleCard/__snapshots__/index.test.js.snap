// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spacex-shuttle-card"
    data-testid="shuttle-card"
  >
    <div
      class="spacex-shuttle-card__image d-flex justify-content-center"
    >
      <img
        alt="FalconSat"
        class="spacex-shuttle-card__image--img"
        src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
      />
    </div>
    <h3
      class="spacex-shuttle-card__name"
    >
      FalconSat
    </h3>
    <div
      class="spacex-shuttle-card__mission-id"
    >
      <strong>
        Mission Ids:
      </strong>
      <ul>
        <li>
          1234
        </li>
      </ul>
    </div>
    <div
      class="spacex-shuttle-card__details"
    >
      <strong>
        Launch Year:
      </strong>
      <span>
        2006
      </span>
    </div>
    <div
      class="spacex-shuttle-card__details"
    >
      <strong>
        Successful Launch:
      </strong>
      <span>
        false
      </span>
    </div>
    <div
      class="spacex-shuttle-card__details"
    >
      <strong>
        Successful Landing:
      </strong>
      <span>
        false
      </span>
    </div>
  </div>
</DocumentFragment>
`;
